### DISCUSSION TITLE
ZST39 Controller Issues: "Timeout while waiting for an ACK" or status of "Unresponsive"

### DISCUSSION THREAD

**Discussion by cjhesse** [zwave-js/zwave-js#6911] [Request Support / Investigate Issue] [answered]
### Checklist

- [X] I have read and followed the above instructions
- [X] I have checked the [troubleshooting section](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index) and my problem is **not** described there.

- [X] I have read the [changelog](https://github.com/zwave-js/node-zwave-js/blob/master/CHANGELOG.md) and my problem was **not** mentioned there or the fix **did not work**.

### Describe the issue

My controller (Zooz ZST39) is only sporadically able to communicate with zwave-js. After removing and reinserting the USB stick, it appears to work for a brief amount of time, but the status becomes "Unresponsive" shortly thereafter. After restarting the zwave-js-ui Docker, the error becomes: Driver: Failed to initialize the driver: ZWaveError: Timeout while waiting for an ACK from the Controller (ZW0200)".

Up until a few days ago, my network was generally pretty reliable with about 35 nodes. I have a Zooz ZST39 that came from the factory with the 7.19.3 firmware. I'd seen the warning about the bug that could cause the stick to become jammed, but never before encountered issues.

A couple nights ago, the stick became jammed for the first time. I think it was resolved after I restarted zwave-js-ui. It happened again last night, so I looked into it a little more. I saw some discussion about newer firmware versions potentially resolving it, so I found 7.21.3 on Zooz's website and installed that using SiLabs PC software. By all appearances, installing the new firmware was successful.

However, since I've installed that firmware, the controller has been virtually unusable. When the stick is first inserted, the driver is started and the stick appears "Ready", but this quickly changes to being intermittently "Unavailable". Commands are very unreliable. As previously mentioned, as soon as I try to restart zwave-js-ui, I get the ACK timeout. It stays this way until I remove and reinsert the stick. No other actions (usbreset, restarting zwave-js-ui, rebooting the server) can get it out of that state.

### Software versions

Driver (node-zwave-js): 12.10.0 (downgrading to 12.9.1, the last version before these errors, did not resolve it)

Z-Wave JS UI: 9.13.4 (downgrading to 9.13.3 did not resolve)

Z-Wave JS UI running in Docker in Unraid 6.12.10.

### Device information

Manufacturer: Zooz
Model name: ZST39

### Checklist

- [X] I [made sure](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index?id=rule-1-use-the-driver-log-luke) to provide a **driver log** on level **debug**.

- [X] The log includes a re-interview of the problematic device _(if applicable)_.

- [X] The log includes the problematic interaction with the device _(if applicable)_.

- [X] I provided the node ID of the problematic device _(if applicable)_.

### Upload Logfile

The log is... very long. But it should contain entries with the stick firmware both on version 7.19.3 and 7.21.3 plus multiple versions of zwave-js/zwave-js-ui.
[zwavejs_2024-06-09.zip](https://github.com/user-attachments/files/15755945/zwavejs_2024-06-09.zip)

**Comment by cjhesse** [comment]
I should have read the update documentation before attempting the firmware upgrade. [Per SiLabs](https://www.silabs.com/documents/public/release-notes/SRN14930-7.21.3.0.pdf), the issue (1227385) isn't even fixed in 7.21.3.

Still... I'd appreciate ideas from anyone to at least get it working again. If I can do that, I can at least try to workaround the lockups by commanding the stick to reset if it gets stuck. But right now it's unusable.

**Comment by kpine** [comment, marked as answer]
Try to downgrade. Already a lengthy discussion at https://github.com/zwave-js/node-zwave-js/discussions/6874.
